client/src/components/AlumniDirectory.js

```diff
@@ -1,79 +1,266 @@
 import React, { Component } from 'react';
-import { Card, Image, Search, Pagination, Grid, Segment } from 'semantic-ui-react'
+import { Card, Image, Search, Pagination, Grid, Segment, Button, Dropdown } from 'semantic-ui-react'
+import { makeCall } from '../apis';
+
+// Filter dropdown options
+const searchOptions = [
+    {
+        key: 'All Fields',
+        text: 'All Fields',
+        value: 'all'
+    },
+    {
+        key: 'Location',
+        text: 'Location',
+        value: 'location'
+    },
+    {
+        key: 'College',
+        text: 'College',
+        value: 'college'
+    },
+    {
+        key: 'Profession',
+        text: 'Profession',
+        value: 'profession'
+    },
+    {
+        key: 'Company',
+        text: 'Company',
+        value: 'company'
+    },
+    {
+        key: 'Name',
+        text: 'Name',
+        value: 'name'
+    },
+    {
+        key: 'Graduation Year',
+        text: 'Graduation Year',
+        value: 'gradYear'
+    }
+]
 
 /*
 props:
-- entries: list of alumni profiles
+- isAlumniView: shows book request button if false
 */
 export default class AlumniDirectory extends Component {
     state = {
         activePage: 1,
-        numEntries: this.props.entries.length,
+        numEntries: 0,
         isLoading: false,
         value: '',
         pageSize: 3,
-        totalPages: Math.ceil(this.props.entries.length/3)
+        totalPages: 0,
+        entries:[],
+        gradYears: [],
+        allText: [],
+        results: 0,
+        filter: 'all'
+    }
+
+    async componentWillMount() {
+        let result = await this.getEntries()
```

Comment:
We can get rid of all awaits I think, apart from this one...
If you inspect `result` returned here, it should not be a PENDING promise.